src/services/data-query.js

```diff
@@ -28,47 +28,78 @@ function QueryService(){
   };
 
   /**
-   *
-   * @param geometry
-   * @param feature_count
-   * @param multilayers
-   * @param condition
-   * @param excludeLayers
+   * @param {{ feature: unknown, feature_count: unknown, filterConfig: unknown, multilayers: boolean, condition: boolean, excludeLayers: unknown[] }}
+   * 
    * @returns {Promise<unknown>}
    */
-  this.polygon = function({feature, feature_count=this.project.getQueryFeatureCount(), filterConfig={}, multilayers=false, condition=this.condition, excludeLayers=[]}={}) {
-    const polygonLayer = excludeLayers[0];
-    const fid = feature.get(G3W_FID);
+  this.polygon = function({
+    feature,
+    feature_count   = this.project.getQueryFeatureCount(),
+    filterConfig    = {},
+    multilayers     = false,
+    condition       = this.condition,
+    /**
+     * @since v3.8
+     */
+    layer    = null,
+    excludeSelected = null
```

Comment:
New parameters have been added:

```js
layer           = null,
excludeSelected = null
```

and the following has been removed:

```js
excludeLayers = []
```

Is this intentional?

Double check if we need to write a fallback condition within the `polygon()` function in order to avoid any new potential unexpected errors.